package/src/components/Channel/Channel.tsx

```diff
@@ -1078,7 +1110,7 @@ const ChannelWithContext = (props: PropsWithChildren<ChannelPropsWithContext>) =
   });
 
   const resyncChannel = useStableCallback(async () => {
-    if (!channel || syncingChannelRef.current) {
+    if (!channel || syncingChannelRef.current || (!channel.initialized && !channel.offlineMode)) {
```

Comment:
This essentially means that:

- If the channel is not `initialized` and in `offlineMode` (fetched from the DB), we should not resync
- If the channel is `initialized` and not in `offlineMode`, we should resync (since we lost connection and regained it)
- If the channel is not `initialized` and not in `offlineMode`, we should **not** resync (this is basically a user opening the `channel` in read-only mode for example, or something of the sort)
- If the channel is both `initialized` and in `offlineMode`, we should resync (it means that we opened the app while offline, regained connection and are now ready to query any out-of-date channels)